compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -394,8 +400,8 @@ fn run_compiler(
                         Ok(())
                     })?;
                 } else {
-                    let krate = queries.parse()?.steal();
-                    pretty::print_after_parsing(sess, &krate, *ppm);
+                    let krate = queries.parse()?;
+                    pretty::print_after_parsing(sess, &krate.borrow(), *ppm);
```

Comment:
I'm really not sure about this change. It was required to not ICE with `-Zunpretty=normal`.

<details>

```
thread 'rustc' panicked at 'attempt to steal from stolen value', compiler/rustc_interface/src/queries.rs:136:43
stack backtrace:
   0:     0x7f5d9b0eb227 - std::backtrace_rs::backtrace::trace_unsynchronized::hfd0a598af8c993aa
   1:     0x7f5d9b0d4dbd - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hd88fdfbb83a6a3c9
   2:     0x7f5d9b16e49f - core::fmt::write::hd2f8cb3b2beeb8ee
   3:     0x7f5d9b0e9131 - std::io::Write::write_fmt::hd3a3b3d29cccc9ac
   4:     0x7f5d9b0d4bf2 - std::sys_common::backtrace::print::h0026f7f026b46fff
   5:     0x7f5d9b0ed640 - std::panicking::panic_hook_with_disk_dump::{{closure}}::hef4f884ecc96e810
   6:     0x7f5d9b0ed43e - std::panicking::panic_hook_with_disk_dump::h170ddc6c9a669136
   7:     0x7f5d9bba251d - rustc_driver_impl[238b128b628687dc]::install_ice_hook::{closure#0}
   8:     0x7f5d9b0edaf2 - std::panicking::rust_panic_with_hook::h4a906cb14242cb6e
   9:     0x7f5d9b0d6902 - std::panicking::begin_panic_handler::{{closure}}::hd77ad73ba114dc3a
  10:     0x7f5d9b0d6836 - std::sys_common::backtrace::__rust_end_short_backtrace::hbb10f9364fe7ff3c
  11:     0x7f5d9b0ed7b2 - rust_begin_unwind
  12:     0x7f5d9b0c64e3 - core::panicking::panic_fmt::hfe6d8bfd6c39e862
  13:     0x7f5d9b0c6cb3 - core::option::expect_failed::ha3e30bd1e33cd88b
  14:     0x7f5d9bd41e99 - <rustc_data_structures[82146c476381c2bd]::steal::Steal<rustc_ast[36ba68c6b808fde4]::ast::Crate>>::steal
  15:     0x7f5d9bd35b4f - <rustc_interface[212019a9a684c909]::queries::Query<(rustc_ast[36ba68c6b808fde4]::ast::Crate, thin_vec[c3b1a873f1736dcf]::ThinVec<rustc_ast[36ba68c6b808fde4]::ast::Attribute>)>>::compute::<<rustc_interface[212019a9a684c909]::queries::Queries>::pre_configure::{closure#0}>
  16:     0x7f5d9bd353c8 - <rustc_interface[212019a9a684c909]::queries::Query<rustc_span[4554545a6fc8d63]::symbol::Symbol>>::compute::<<rustc_interface[212019a9a684c909]::queries::Queries>::crate_name::{closure#0}>
  17:     0x7f5d9bd35532 - <rustc_interface[212019a9a684c909]::queries::Query<&rustc_middle[ce4b6cf446d87f8a]::ty::context::GlobalCtxt>>::compute::<<rustc_interface[212019a9a684c909]::queries::Queries>::global_ctxt::{closure#0}>
  18:     0x7f5d9bc1859d - <rustc_interface[212019a9a684c909]::interface::Compiler>::enter::<rustc_driver_impl[238b128b628687dc]::run_compiler::{closure#1}::{closure#2}, core[4752e992e8614a12]::result::Result<core[4752e992e8614a12]::option::Option<rustc_interface[212019a9a684c909]::queries::Linker>, rustc_span[4554545a6fc8d63]::ErrorGuaranteed>>
  19:     0x7f5d9bbf1ee0 - rustc_span[4554545a6fc8d63]::set_source_map::<core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>, rustc_interface[212019a9a684c909]::interface::run_compiler<core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>, rustc_driver_impl[238b128b628687dc]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
  20:     0x7f5d9bba67f5 - <scoped_tls[c42a965abf1356a6]::ScopedKey<rustc_span[4554545a6fc8d63]::SessionGlobals>>::set::<rustc_interface[212019a9a684c909]::interface::run_compiler<core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>, rustc_driver_impl[238b128b628687dc]::run_compiler::{closure#1}>::{closure#0}, core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>>
  21:     0x7f5d9bc1e680 - std[e15f0322381c82ac]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[212019a9a684c909]::util::run_in_thread_pool_with_globals<rustc_interface[212019a9a684c909]::interface::run_compiler<core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>, rustc_driver_impl[238b128b628687dc]::run_compiler::{closure#1}>::{closure#0}, core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>>
  22:     0x7f5d9bc1ad46 - <core[4752e992e8614a12]::panic::unwind_safe::AssertUnwindSafe<<std[e15f0322381c82ac]::thread::Builder>::spawn_unchecked_<rustc_interface[212019a9a684c909]::util::run_in_thread_pool_with_globals<rustc_interface[212019a9a684c909]::interface::run_compiler<core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>, rustc_driver_impl[238b128b628687dc]::run_compiler::{closure#1}>::{closure#0}, core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>>::{closure#1}::{closure#0}> as core[4752e992e8614a12]::ops::function::FnOnce<()>>::call_once
  23:     0x7f5d9bc16f54 - std[e15f0322381c82ac]::panicking::try::<core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>, core[4752e992e8614a12]::panic::unwind_safe::AssertUnwindSafe<<std[e15f0322381c82ac]::thread::Builder>::spawn_unchecked_<rustc_interface[212019a9a684c909]::util::run_in_thread_pool_with_globals<rustc_interface[212019a9a684c909]::interface::run_compiler<core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>, rustc_driver_impl[238b128b628687dc]::run_compiler::{closure#1}>::{closure#0}, core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>>::{closure#1}::{closure#0}>>
  24:     0x7f5d9bc1f97a - <<std[e15f0322381c82ac]::thread::Builder>::spawn_unchecked_<rustc_interface[212019a9a684c909]::util::run_in_thread_pool_with_globals<rustc_interface[212019a9a684c909]::interface::run_compiler<core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>, rustc_driver_impl[238b128b628687dc]::run_compiler::{closure#1}>::{closure#0}, core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[4752e992e8614a12]::result::Result<(), rustc_span[4554545a6fc8d63]::ErrorGuaranteed>>::{closure#1} as core[4752e992e8614a12]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  25:     0x7f5d9b11fd59 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h83ec483d9dee03e2
  26:     0x7f5d9b0e7f4a - std::sys::unix::thread::Thread::new::thread_start::hedaabae35861c40f
  27:     0x7f5d9aeee44b - <unknown>
  28:     0x7f5d9af71e40 - <unknown>
  29:                0x0 - <unknown>
```

</details>